src/compiler/emitter.ts

```diff
@@ -4395,7 +4413,11 @@ module ts {
                     emitComputedPropertyName(<ComputedPropertyName>memberName);
                 }
                 else {
-                    write(".");
+                    // For script-target that is ES6 or above, we want to emit memberName by itself without prefix "." if the memberName is a name of method.

```

Comment:
yes. I realize from reading the code now that it is hard to understand what this code tries to do